src/main/java/net/sf/jabref/gui/maintable/MainTable.java

```diff
@@ -499,11 +499,14 @@ public void setSelected(int row) {
     }
 
     public int findEntry(BibEntry entry) {
-        return model.getTableRows().indexOf(entry);
-    }
-
-    public int findLastEntry(BibEntry entry) {
-        return model.getTableRows().lastIndexOf(entry);
+        EventList<BibEntry> tableRows = model.getTableRows();
+        for (int row = 0; row < tableRows.size(); row++) {
+            BibEntry bibEntry = tableRows.get(row);
+            if (entry.getId().equals(bibEntry.getId())) {
```

Comment:
I don't like that the ID is used to compare the entries (because I hope that we can get rid of the ID completely at some point). Does `if (entry == bibentry)` work?